### PR TITLE
Feature: Multi-tile docks.

### PR DESCRIPTION
**Pull request by PeterN**
* Multiple docks can be built in one station.
* Any water tile adjacent to a dock can be used for docking, not just the tile at the end of the dock.
* Any water tile surrounding an industry with a dock can be used for docking.

m1 bit 7 is (ab)used to mark docking tiles. These can be on MP_WATER, MP_RAILWAY (half-tile on water), MP_STATION (buoys) and MP_TUNNELBRIDGE (aqueduct ends)

### CODE REVIEW

**Comment by PeterN** [comment]
Found one issue, if there's a suitable tile _behind_ the dock, ships could also dock there. This may be desirable in the future but is not for the current style of dock, I think.

**Comment by andythenorth** [comment]
Testing this in a game currently.  No issues so far.

File: src/pathfinder/npf/npf.cpp [hunk 1]
```diff
@@ -319,6 +329,13 @@ static int32 NPFWaterPathCost(AyStar *as, AyStarNode *current, OpenListNode *par
 		cost += _settings_game.pf.npf.npf_water_curve_penalty;
 	}
 
+	if (IsDockingTile(current->tile)) {
+		/* Check docking tile for occupancy */
+		uint count = 1;
+		HasVehicleOnPos(current->tile, &count, &CountShipProc);
+		cost += count * 3 * _trackdir_length[trackdir];
```
Comment (LordAro) [inline src/pathfinder/npf/npf.cpp:336]: why `* 3` ?

**Review by LordAro — APPROVED** [review]
Sure, why not

**Pull request merged**